### PR TITLE
Split off static libararies on unix

### LINKED ISSUE #76: Split off static builds
It seems that in 
https://github.com/conda-forge/fftw-feedstock/pull/50
A discussion started on whether or not we should include the static files.

I believe conda-forge has since pushed away from static libraries.

Would it be acceptable to remove the static libraries from this package and deal with the issues when they arise downstream?
Can we split the package instead into two?
```
fftw-static   # not recommended but for downstream projects to migrate
fftw  # only dynamic modules
```

Thoughts?

### PR DESCRIPTION
**Pull request by hmaarrfk**
Closes #76
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@conda-forge-admin please rerender

This is ready for review.

**Comment by hmaarrfk** [comment]
Lol. This used to work....

**Comment by egpbos** [comment]
It's a bit hard to review like this. Could you please squash your commits into one? I can also do it for you if you prefer.

**Comment by hmaarrfk** [comment]
commit history should be cleaner now.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
-{% set build = 3 %}
+{% set build = 4 %}
 
 
 # ensure mpi is defined (needed for conda-smithy recipe-lint)
 {% set mpi = mpi or 'nompi' %}
 
 package:
-  name: fftw
+  name: fftw-split
```
Comment (egpbos) [inline recipe/meta.yaml:9]: Does it have to be a different name? I'm not sure if it matters, but just wondering whether changing the name will clash with any bot automation.

File: recipe/test_cmake.sh [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env bas
-cd ${RECIPE_DIR}/test_cmake
+cd test_cmake
```
Comment (egpbos) [inline recipe/test_cmake.sh:2]: Was this necessary? Seems like a different issue? I guess if it doesn't hurt...

File: recipe/meta.yaml [hunk 3]
```diff
@@ -21,123 +20,186 @@ build:
...
+        - test_cmake/*            # [not win]
+      requires:
+        - python >=3
+        - cmake                   # [not win]
+        - make                    # [not win]
+        - {{ compiler('c') }}     # [not win]
+        - {{ compiler('cxx') }}   # [not win]
+
+      commands:
+        # Verify the non existence of library files
```
Comment (egpbos) [inline recipe/meta.yaml:80]:
```suggestion
        # Verify that static library files aren't present yet
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -21,123 +20,186 @@ build:
...
+      run:
+        - llvm-openmp  # [osx]
+        - {{ pin_subpackage('fftw', exact=True) }}
+
+    test:
+      commands:
+        # Verify library contains Fortran symbols
+        - strings ${PREFIX}/lib/libfftw3.a | grep -q dfftw      # [not win]
+
+        # Verify existence of library files
```
Comment (egpbos) [inline recipe/meta.yaml:184]:
```suggestion
        # Verify existence of static library files
```

**Review by egpbos — CHANGES_REQUESTED** [review]
This looks great! I just have a few minor suggestions. I think after you addressed those we can merge.

File: recipe/test_cmake.sh (hunk 2 above)
Comment (hmaarrfk) [reply recipe/test_cmake.sh:2]: I was confused about this too. I think there are intricacies between how the recipe dir is accessed for split recipes.

I had to declare `test_cmake` to be copied in from the recipe dir at test time.

File: recipe/meta.yaml (hunk 1 above)
Comment (hmaarrfk) [reply recipe/meta.yaml:9]: There is some inconsistent behavior in how conda treats split packages where the top level package has the same name as one of the outputs.

I guess caution has made people just call it something else.

**Comment by egpbos** [comment]
Awesome, thanks for this great PR!

**Comment by hmaarrfk** [comment]
For some reason, the build number increment didnt' work out:
image

**Comment by hmaarrfk** [comment]
the yare both build 4, one should be 104

**Pull request merged**